examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/AppPlatformService.java

```diff
@@ -98,17 +118,39 @@ public void setActivity(Activity activity) {
   }
 
   public void addContentApp(ContentApp app) {
-    app.setEndpointId(
-        mAppPlatform.addContentApp(
-            app.getVendorName(),
-            app.getVendorId(),
-            app.getAppName(),
-            app.getProductId(),
-            "1.0",
-            new ContentAppEndpointManagerImpl(context)));
+    int retEndpointId = -1;
+    int desiredEndpointId = discoveredEndpoints.getInt(app.getAppName(), -1);
+    if (desiredEndpointId > 0) {
+      retEndpointId =
+          mAppPlatform.addContentAppAtEndpoint(
+              app.getVendorName(),
+              app.getVendorId(),
+              app.getAppName(),
+              app.getProductId(),
+              "1.0",
+              desiredEndpointId,
+              new ContentAppEndpointManagerImpl(context));
+    } else {
+      retEndpointId =
+          mAppPlatform.addContentApp(
+              app.getVendorName(),
+              app.getVendorId(),
+              app.getAppName(),
+              app.getProductId(),
+              "1.0",
+              new ContentAppEndpointManagerImpl(context));
+    }
+    if (retEndpointId > 0) {
+      app.setEndpointId(retEndpointId);
+      discoveredEndpoints.edit().putInt(app.getAppName(), app.getEndpointId()).apply();
+    }
   }
 
-  public int removeContentApp(int endpointID) {
-    return mAppPlatform.removeContentApp(endpointID);
+  public int removeContentApp(int endpointID, String appName) {
```

Comment:
Why are both endpointid and appName needed? Are they not known in advance and equivalent - if I know the endpointID should I not know the app name on that endpoint already? 

Please add some documentation about why both are passed and what happens if they do not match.

Reply:
Based on the [TODO](https://github.com/project-chip/connectedhomeip/pull/22354/files/5534efb983342385556df870e15c9a06c1fc68d6#diff-78f855fb33f255484885e18b9af8c228261872b3273b6c041d19d8a11f60537dR77) this code will be removed as we will not be removing endpoints once added.